src/main/java/service/OptionCli.java

```diff
@@ -0,0 +1,113 @@
+                .desc("Print this message")
+                .required(false)
+                .build());
+
+        this.options.addOption(Option.builder("C")
+                .longOpt("current")
+                .hasArg(false)
+                .desc("Set this to view the current weather")
+                .required(false)
+                .build());
+
+        this.options.addOption(Option.builder("F")
+                .longOpt("forecast")
+                .argName(" \"yyyy-MM-dd HH:mm:ss\" ")
+                .hasArg(true)
+                .optionalArg(true)
+                .desc("Set this to view the forecast *no parameter will display available dates and forecast times")
+                .required(false)
+                .build());
+
+        this.options.addOption(Option.builder("c")
+                .longOpt("city")
+                .argName(" name ")
+                .hasArg(true)
+                .desc("Enter the city name")
+                .required(false)
+                .build());
+
+        this.options.addOption(Option.builder("g")
+                .longOpt("geo")
+                .argName(" latitude longitude ")
+                .hasArg(true)
+                .numberOfArgs(2)
+                .desc("Search by geographic coordinates")
+                .required(false)
+                .build());
+
+        return this.options;
+    }
+
+    public void helpBuild() {
+        HelpFormatter formatter = new HelpFormatter();
+        formatter.printHelp("Options ", optionsBuild(), true);
+    }
+
+    public String optionBuildQuery() {
+        //Current
+        if (cmd.hasOption("C") && cmd.hasOption("c")) {
+            StringBuilder stringBuilder = new StringBuilder(cmd.getOptionValue("c"));
+            for (String arg : cmd.getArgList()) {
+                stringBuilder.append("+").append(arg);
+            }
+            return new QueryUrlFactory("C").buildQueryByNameCity(stringBuilder.toString());
+        }
+
+        if (cmd.hasOption("C") && cmd.hasOption("g")) {
+            List<String> values = Arrays.asList(cmd.getOptionValues("g"));
+            return new QueryUrlFactory("C").buildQueryByCoord(values.get(0), values.get(1));
+        }
+
+        //Forecast
+        if (cmd.hasOption("F") && cmd.hasOption("c")) {
+            this.dataTime = cmd.getOptionValue("F");
+            StringBuilder stringBuilder = new StringBuilder(cmd.getOptionValue("c"));
+            for (String arg : cmd.getArgList()) {
+                stringBuilder.append("+").append(arg);
+            }
+            return new QueryUrlFactory("F").buildQueryByNameCity(stringBuilder.toString());
+        }
+
+        if (cmd.hasOption("F") && cmd.hasOption("g")) {
+            this.dataTime = cmd.getOptionValue("F");
+            List<String> values = Arrays.asList(cmd.getOptionValues("g"));
+            return new QueryUrlFactory("F").buildQueryByCoord(values.get(0), values.get(1));
+        }
+        return Arrays.toString(this.arguments);
```

Comment:
Jeśli ktoś podał niewlaściwą kombinacje parametrów to rzuć wyjątek zamiast zwracać coś co spowoduje błąd poźniej